source/particle/world.cc

```diff
@@ -391,11 +391,19 @@ namespace aspect
                 else if ((particle_load_balancing & ParticleLoadBalancing::remove_particles) &&
                          (n_particles_in_cell > max_particles_per_cell))
                   {
+                    const unsigned int n_particles_to_remove = n_particles_in_cell - max_particles_per_cell;
+
+#if DEAL_II_VERSION_GTE(10,0,0)
+                    for (unsigned int i=0; i < n_particles_to_remove; ++i)
+                      {
+                        auto particle_to_remove = particle_handler->particles_in_cell(cell).begin();
+                        std::advance(particle_to_remove, random_number_generator() % particle_handler->n_particles_in_cell(cell));
```

Comment:
This relies on the assumption that the `random_number_generator()` call returns an integer equally distributed between zero and a multiple of `particle_handler->n_particles_in_cell(cell)`, but that's not necessarily true. It could just return zero or one in each call, or 0...127, which is not what you want if you had, for example, 200 particles.

The generic way to do what you want is to use `std::uniform_int_distribution` and let it give you a random integer in the range you want (inclusive):
```suggestion
                        std::advance(particle_to_remove, 
                                     std::uniform_int_distribution<unsigned int>(0, particle_handler->n_particles_in_cell(cell)-1)(random_number_generator());
```